exercises/tournament/example.rs

```diff
@@ -77,13 +67,12 @@ fn write_tally(results: &HashMap<String, TeamResult>, output_filename: &Path) ->
                   Equal => a.1.cmp(&(b.1)).reverse(),
                   other => other
               });
-    let mut f = try!(File::create(output_filename));
-    try!(writeln!(&mut f, "{:30} | MP |  W |  D |  L |  P", "Team"));
-    for &(ref team, games, r, points) in v.iter() {
-        try!(writeln!(&mut f, "{:30} | {:2} | {:2} | {:2} | {:2} | {:2}",
-                      team, games, r.wins, r.draws, r.losses, points));
-    }
-    Ok(())
+    let header = format!("{:30} | MP |  W |  D |  L |  P\n", "Team");
+    let lines: Vec<_> = v.iter().map(|&(ref team, games, r, points)| {
+        format!("{:30} | {:2} | {:2} | {:2} | {:2} | {:2}",
+                team, games, r.wins, r.draws, r.losses, points)
+    }).collect();
+    header + &lines.join("\n")
```

Comment:
Note that for an empty input, this prints the header and nothing after. No test tests this behavior, so I have nothing to say about whether this appropriate.